### PR TITLE
Improved logging

### PR DESCRIPTION
**Pull request by tino1b2be**
Removed the old logging system "twisted.python.log" and replaced it with the new twisted logging system which uses "twisted.logger.Logger".
Added more logging messages for debugging and general logs.

### CODE REVIEW

**Comment by tino1b2be** [comment]
I've added quite a number of log entries in service.py but not so many in util.py (Don't really know what events to log). Are there any other places/events that need to be logged?

File: docker_xylem/service.py [hunk 1]
```diff
@@ -36,6 +37,7 @@ def __init__(self, config):
         self.current = {}
 
     def xylem_request(self, queue, call, data):
+        self.log.info('HTTP request to xylem.')
```
Comment (jerith) [inline docker_xylem/service.py]: Maybe we should add more detail here? The volume name seems like a good thing to know.

File: docker_xylem/service.py [hunk 2]
```diff
@@ -79,16 +81,22 @@ def _umount_fs(self, path):
...
             # Return true is mounted
+            self.log.info('Successfully unmounted {path}', path=path)
             defer.returnValue(True)
 
     @defer.inlineCallbacks
     def mount_volume(self, request, data):
+        self.log.debug(
+            'Function Call: mount_volume(), `data` = \"{data}\"',
+            data=str(data)
+        )
```
Comment (jerith) [inline docker_xylem/service.py]: Should we maybe log this in the request dispatcher rather than each method?

File: docker_xylem/service.py [hunk 3]
```diff
@@ -140,25 +174,38 @@ def get_volume_path(self, request, data):
...
         name = data['Name']
 
         result = yield self.xylem_request('gluster', 'createvolume', {
             'name': name
         })
 
         if not result['result']['running']:
+            self.log.error('Error creating volume {name}', name=name)
```
Comment (jerith) [inline docker_xylem/service.py]: It would be nice to have more detail here. Do we get anything useful from xylem?

File: docker_xylem/service.py [hunk 4]
```diff
@@ -213,14 +272,18 @@ def _route_request(self, request):
         method = self.requestRouter.get(request.path)
 
         if not method:
+            self.log.warn(
+                '{request.path} is not implemented by plugin.',
+                request=request
+            )
             return "Not Implemented"
```
Comment (jerith) [inline docker_xylem/service.py:249]: We probably want to return HTTP 404 or something here. Does the docker volume driver documentation say anything about what docker expects?

(Note: This isn't about the logging. If we do decide to change the behaviour, let's open a separate ticket for that.)

File: docker_xylem/service.py (hunk 1 above)
Comment (tino1b2be) [reply docker_xylem/service.py]: Cool I didn't realise this function was only used when creating a volume.

File: docker_xylem/service.py (hunk 4 above)
Comment (tino1b2be) [reply docker_xylem/service.py:249]: Docker's docs don't specify what response it expects if a plugin does not implement a certain feature.

File: docker_xylem/service.py (hunk 2 above)
Comment (tino1b2be) [reply docker_xylem/service.py]: Yes that's a better idea. Will do that.

File: docker_xylem/service.py (hunk 3 above)
Comment (tino1b2be) [reply docker_xylem/service.py]: Xylem only returns the name of the volume and the ID. There is no information on any errors.

**Review by jerith — APPROVED** [review]
:+1:

**Pull request merged**